lua/entities/gmod_wire_expression2/core/custom/cfc_vector2_NumberOperands.lua

```diff
@@ -0,0 +1,537 @@
+E2Lib.RegisterExtension( "cfc_e2_lib", true )
+
+-- these upvalues (locals in an enclosing scope) are faster to access than globals.
+local delta  = wire_expression2_delta
+
+local math   = math
+local random = math.random
+local pi     = math.pi
+local inf    = math.huge
+
+local exp    = math.exp
+local frexp  = math.frexp
+local log    = math.log
+local log10  = math.log10
+local sqrt   = math.sqrt
+local abs    = math.abs
+
+local floor  = math.floor
+local ceil   = math.ceil
+local Round  = math.Round
+
+local sin    = math.sin
+local cos    = math.cos
+local tan    = math.tan
+
+local acos   = math.acos
+local asin   = math.asin
+local atan   = math.atan
+
+__e2setcost(2) -- approximation
+
+--- Returns true (1) if given value is a finite number; otherwise false (0).
+e2function vector2 isfinite(vector2 vectah)
+    return {
+        (vectah[1] > -inf and vectah[1] < inf) and 1 or 0,
+        (vectah[2] > -inf and vectah[2] < inf) and 1 or 0
+    }
+end
+
+e2function vector2 vector2:isfinite()
+    return {
+        (this[1] > -inf and this[1] < inf) and 1 or 0,
+        (this[2] > -inf and this[2] < inf) and 1 or 0
+    }
+end
+
+--- Returns 1 if given value is a positive infinity or -1 if given value is a negative infinity; otherwise 0.
+e2function vector2 isinf(vector2 vectah)
+    local function numIsInf(number)
```

Comment:
```suggestion
    local function numIsInf(number)
```

Helper functions go at the top of the file

Reply:
Done